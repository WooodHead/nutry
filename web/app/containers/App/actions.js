import {
  SEARCH_COMPLETE,
} from 'containers/Search/constants';

import {
  DEFAULT_ACTION,
  LOGIN_CLICKED,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
} from './constants';


export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}
export function login() {
  return {
    type: LOGIN_CLICKED,
  };
}
export function loginSuccess(token) {
  return {
    type: LOGIN_SUCCESS,
    token,
  };
}

export function searchComplete(searchResults) {
  return {
    type: SEARCH_COMPLETE,
    searchResults,
  };
}

export function loginFailure(err) {
  return {
    type: LOGIN_FAILURE,
    err,
  };
}
