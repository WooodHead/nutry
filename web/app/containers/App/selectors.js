import { createSelector } from 'reselect';

const selectRoute = (state) => state.get('route');
const selectGlobal = (state) => state.get('global');

const makeSelectLocation = () => createSelector(
  selectRoute,
  (routeState) => routeState.get('location').toJS()
);
const makeSelectLoggedIn = () => createSelector(
  selectGlobal,
  (globalState) => globalState.get('loggedIn')
);
const makeSelectUsername = () => createSelector(
  selectGlobal,
  (globalState) => globalState.get('username')
);
const makeSelectSearchResults = () => createSelector(
  selectGlobal,
  (globalState) => globalState.get('searchResults')
);
const makeSelectSearchLoading = () => createSelector(
  selectGlobal,
  (globalState) => globalState.get('loading')
);

export {
  makeSelectLocation,
  makeSelectLoggedIn,
  makeSelectUsername,
  makeSelectSearchResults,
  makeSelectSearchLoading,
};
