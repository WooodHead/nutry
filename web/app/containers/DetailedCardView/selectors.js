import { createSelector } from 'reselect';

/**
 * Direct selector to the detailedCardView state domain
 */
const selectDetailedCardViewDomain = (state) => state.get('foodProfile');

/**
 * Other specific selectors
 */


/**
 * Default selector used by DetailedCardView
 */
const makeSelectNutrientBySection = () => createSelector(
  selectDetailedCardViewDomain,
  (substate) => substate.getIn(['nutrients', 'bySection'])
);
export {
  makeSelectNutrientBySection,
};
