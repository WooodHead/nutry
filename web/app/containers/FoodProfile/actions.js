/*
 *
 * FoodProfile actions
 *
 */
import { defaultPortions } from 'lib/nutrientMap';
import {
  DEFAULT_ACTION,
  ON_BACK,
  GET_PROFILE,
  TAB_CHANGED,
  PORTION_CHANGED,
  AGE_GROUP_CHANGED,
  NUTRIENT_SELECTED,
  GET_PROFILE_FAILURE,
  GET_PROFILE_SUCCESS,
} from './constants';


export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}
export function loadProfile(serialNumber, source) {
  return {
    type: GET_PROFILE,
    serialNumber,
    source,
  };
}
export function loadProfileSuccess(profileBody) {
  return {
    type: GET_PROFILE_SUCCESS,
    nutrientsById: profileBody.nutrients,
    portionsAvailable: defaultPortions(profileBody.portions),
  };
}
export function loadProfileFailure(error) {
  return {
    type: GET_PROFILE_FAILURE,
    error,
  };
}
export function tabChanged(tab) {
  return {
    type: TAB_CHANGED,
    tab,
  };
}
export function nutrientSelected(prefix, id) {
  return {
    type: NUTRIENT_SELECTED,
    prefix,
    id,
  };
}
export function portionChanged(portionSelected) {
  return {
    type: PORTION_CHANGED,
    portionSelected,
  };
}
export function ageGroupChanged(ageGroupSelected) {
  return {
    type: AGE_GROUP_CHANGED,
    ageGroupSelected,
  };
}

export function onBack() {
  return {
    type: ON_BACK,
  };
}
export function clearFoodProfile() {
  return {
    type: ON_BACK,
  };
}
