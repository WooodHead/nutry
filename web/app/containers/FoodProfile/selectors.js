import { createSelector } from 'reselect';

/**
 * Direct selector to the foodProfile state domain
 */
const selectFoodProfileDomain = (state) => state.get('foodProfile');

/**
 * Other specific selectors
 */


/**
 * Default selector used by FoodProfile
 */

const makeSelectFoodProfile = () => createSelector(
  selectFoodProfileDomain,
  (substate) => substate.toJS()
);

export default makeSelectFoodProfile;
export {
  selectFoodProfileDomain,
};
