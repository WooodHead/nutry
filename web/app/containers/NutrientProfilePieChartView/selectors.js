import { createSelector } from 'reselect';

const selectFoodProfileDomain = (state) => state.get('foodProfile');


const makeSelectLoading = () => createSelector(
  selectFoodProfileDomain,
  (substate) => substate.get('loading')
);
const makeSelectSummaryNutrients = () => createSelector(
  selectFoodProfileDomain,
  (substate) => substate.getIn(['nutrients', 'bySummaryPie']).toJS()
);
const makeSelectDetailedNutrients = () => createSelector(
  selectFoodProfileDomain,
  (substate) => substate.getIn(['nutrients', 'byDetailedPie']).toJS()
);
const makeSelectNutrients = () => createSelector(
  selectFoodProfileDomain,
  (substate) => substate.get('tabSelected') === 'summary' ?
    substate.getIn(['nutrients', 'bySummaryPie']).toJS() :
    substate.getIn(['nutrients', 'byDetailedPie']).toJS()
);
const makeSelectNutrientSelected = () => createSelector(
  selectFoodProfileDomain,
  (substate) => substate.get('nutrientSelected')
);
const makeSelectPortionSelected = () => createSelector(
  selectFoodProfileDomain,
  (substate) => substate.get('portionSelected')
);
export {
  makeSelectLoading,
  makeSelectNutrients,
  makeSelectSummaryNutrients,
  makeSelectDetailedNutrients,
  makeSelectNutrientSelected,
  makeSelectPortionSelected,
};
