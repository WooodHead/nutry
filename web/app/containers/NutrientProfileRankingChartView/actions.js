/*
 *
 * NutrientProfileRankingChartView actions
 *
 */

import {
  GET_NUTRIENT_RANKING,
  GET_NUTRIENT_RANKING_SUCCESS,
  GET_NUTRIENT_RANKING_FAILURE,
} from './constants';

export function loadRankings() {
  return {
    type: GET_NUTRIENT_RANKING,
  };
}
export function loadRankingsSuccess(rankingResults) {
  return {
    type: GET_NUTRIENT_RANKING_SUCCESS,
    rankingResults,
  };
}
export function loadRankingsFailure(err) {
  return {
    type: GET_NUTRIENT_RANKING_FAILURE,
    err,
  };
}
