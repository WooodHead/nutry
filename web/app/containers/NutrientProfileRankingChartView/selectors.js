import { createSelector } from 'reselect';

/**
 * Direct selector to the nutrientProfileRankingChartView state domain
 */
const selectSearchDomain = (state) => state.get('search');
const selectFoodProfileDomain = (state) => state.get('foodProfile');
const selectNutrientProfileRankingChartViewDomain = (state) => state.get('nutrientRanking');

/**
 * Other specific selectors
 */
const makeSelectSearchResults = () => createSelector(
  selectSearchDomain,
  (substate) => substate.get('results')
);

/**
 * Default selector used by NutrientProfileRankingChartView
 */

const makeSelectNutrientProfileRankingChartView = () => createSelector(
  selectNutrientProfileRankingChartViewDomain,
  (substate) => substate.toJS()
);
const makeSelectLoading = () => createSelector(
  selectNutrientProfileRankingChartViewDomain,
  (substate) => substate.get('loading')
);
const makeSelectRankingResults = () => createSelector(
  selectNutrientProfileRankingChartViewDomain,
  (substate) => substate.get('rankingResults')
);
const makeSelectNutrientSelected = () => createSelector(
  selectFoodProfileDomain,
  (substate) => substate.get('nutrientSelected')
);

export default makeSelectNutrientProfileRankingChartView;
export {
  selectNutrientProfileRankingChartViewDomain,
  makeSelectLoading,
  makeSelectRankingResults,
  makeSelectNutrientSelected,
  makeSelectSearchResults,
};
