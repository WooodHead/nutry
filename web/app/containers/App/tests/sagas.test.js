import { testSaga, expectSaga } from 'redux-saga-test-plan';
// import { expectSaga } from 'redux-saga-test-plan';
import { select, takeLatest, put } from 'redux-saga/effects';
import { delay } from 'redux-saga';

import {
  LOGIN_CLICKED,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
} from '../constants';
import appReducer from '../reducer';
import { makeSelectUsername } from '../selectors';
import { loginSuccess } from '../actions';
import loginData, { loginUser } from '../sagas';


describe('Container [App] - sagas', () => { /* eslint-disable redux-saga/yield-effects */
  describe('loginUser Saga', () => {
    let loginUserGenerator;
    beforeEach(() => {
      loginUserGenerator = loginUser();
      // select username
      const selectDescriptor = loginUserGenerator.next().value;
      console.log(selectDescriptor);
      expect(selectDescriptor).toMatchSnapshot();
      // call delay effect
      loginUserGenerator.next();
      const callDescriptor = loginUserGenerator.next().value;
      console.log(callDescriptor);
    });
    it('should dispatch loginSuccess action if the token is recieved and stored successfully', () => {
      const token = 'yummy-chocolate';
      const putDescriptor = loginUserGenerator.next().value;
      expect(putDescriptor).toEqual(put(loginSuccess(token)));
    });
  });
  describe('loginData Saga', () => {
    const loginDataSaga = loginData();
    it('works with unit tests', () => {
      const takeLatestDescriptor = loginDataSaga.next().value;
      expect(takeLatestDescriptor).toEqual(takeLatest(LOGIN_CLICKED, loginUser));
    });
  });
});
