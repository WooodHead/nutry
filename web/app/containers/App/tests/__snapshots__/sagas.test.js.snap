// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container [App] - sagas loginUser Saga should dispatch loginSuccess action if the token is recieved and stored successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;
